mito-ai/mito_ai/gemini_client.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GNU Affero General Public License v3.0 License.
+
+from typing import AsyncGenerator
+from google import genai
+from mito_ai.completions.models import AgentResponse
+
+
+class GeminiClient:
+    def __init__(self, api_key: str, model: str):
+        self.client = genai.Client(api_key=api_key)
+        self.model = model
+
+    async def generate_content(self, contents: str) -> str:
+        try:
+            response = self.client.models.generate_content(
+                model=self.model,
+                contents=contents,
+                config={
+                "response_mime_type": "application/json",
+                "response_schema": AgentResponse
+                }
+            )
+            
+            if not response:
+                return "No response received from Gemini API"
+                
+            if hasattr(response, 'text') and response.text:
+                return response.text
+
+            # Handle streaming response
```

Comment:
When we talked through this code there was some good additional context you shared about why we needed this code. Can you add extra context like that as comments here. I already forget what you told me, and it would be required info if I ever needed to refactor this code.